test/e2e/gc_e2e_test.go

```diff
@@ -33,159 +34,243 @@ var _ = Describe("Garbage collector", func() {
+							Name: "dependent",
+						},
+						Data: map[string]string{},
+					}
 
-		// delete ownerB in the foreground (to ensure any "blocking" dependents are deleted before ownerB)
-		err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(fetchedB.GetName(), &options)
-		require.NoError(GinkgoT(), err)
+					// add owners
+					ownerutil.AddOwner(dependent, fetchedA, true, false)
+					ownerutil.AddOwner(dependent, fetchedB, true, false)
+
+					// create dependent
+					_, err = kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Create(dependent)
+					Expect(err).NotTo(HaveOccurred(), "dependent could not be created")
+				})
+
+				When("one of the owner is deleted", func() {
+					var (
+						propagation metav1.DeletionPropagation
+						options     metav1.DeleteOptions
+					)
+
+					BeforeEach(func() {
+						propagation = "Foreground"
+						options = metav1.DeleteOptions{PropagationPolicy: &propagation}
+					})
+
+					It("delete ownerA in the foreground", func() {
+
+						// delete ownerA in the foreground (to ensure any "blocking" dependents are deleted before ownerA)
+						err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(fetchedA.GetName(), &options)
+						Expect(err).NotTo(HaveOccurred())
+
+						// wait for deletion of ownerA
+						waitForDelete(func() error {
+							_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(ownerA.GetName(), metav1.GetOptions{})
+							return err
+						})
+
+						//By("check for dependent (should still exist since it still has one owner present", func() {
+						// check for dependent (should still exist since it still has one owner present)
+						cf, err := kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Get(dependent.GetName(), metav1.GetOptions{})
+						fmt.Printf("###A %v", cf)
+						Expect(err).NotTo(HaveOccurred(), "dependent deleted after one owner was deleted")
+						ctx.Ctx().Logf("dependent still exists after one owner was deleted")
+
+					})
+
+					It("delete ownerB in the foreground", func() {
+						// delete ownerB in the foreground (to ensure any "blocking" dependents are deleted before ownerB)
+						err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(fetchedB.GetName(), &options)
+						Expect(err).NotTo(HaveOccurred())
+
+						// wait for deletion of ownerB
+						waitForDelete(func() error {
+							_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(ownerB.GetName(), metav1.GetOptions{})
+							return err
+						})
+
+						//By("check for dependent (should be deleted since last blocking owner was deleted", func() {
+						// check for dependent (should be deleted since last blocking owner was deleted)
+						cf, err := kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Get(dependent.GetName(), metav1.GetOptions{})
```

Comment:
The test ensures that the dependent `configMap` is not deleted if one of its owner reference is deleted. 

Problem: The assertion at line 266 fails because err is `nil`

What I found: 
When ownerB CSV is deleted, an empty `configMap` struct is returned in the older version. (line 262)
See below output: 
Current version:
```
###B &ConfigMap{ObjectMeta:{dependent  openshift-operators /api/v1/namespaces/openshift-operators/configmaps/dependent 412d8073-d3ce-4d11-9a77-7876f0e8e062 161740 0 2020-04-06 18:25:30 -0400 EDT <nil> <nil> map[] map[] [
```
Older version: 
```
###B &ConfigMap{ObjectMeta:{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Data:map[string]string{},BinaryData:map[string][]byte{},}cleaning up any remaining non persistent resources...
```
The `cf` returned here has value whereas in the older version of the file, `cf` returns an empty ConfigMap struct as seen above

Because of this, at line 266
Current version:
```
 Error is <nil> and value is <nil>
```
vs 
Older version:
```
 Error type *errors.StatusError and value is configmaps "dependent" not found
```

Reply:
I wonder if that has anything todo with your forgrand deletion option. Since the object in the configmap returnd is the dependent.

Reply:
Figured out the issue. It was related to the placement of the deletion code inside of `It` block. Thanks for looking into this!